src/screens/NativeIntegration/NIInputService.tsx

```diff
@@ -0,0 +1,279 @@
+import { useDispatch } from 'react-redux';
+import NIInputField from './components/NIInputField';
+
+// Constant
+import { NI_TRANSACTION_COMPLETED } from 'constants/navigationConstants';
+
+// ABIs
+import ERC20_CONTRACT_ABI from 'abi/erc20.json';
+
+function NIInputService() {
+  const { t } = useTranslation();
+  const navigation = useNavigation();
+  const dispatch = useDispatch();
+  const action = navigation.getParam('action');
+  const contractData = navigation.getParam('contractData');
+  const contractType = navigation.getParam('contractType');
+  const title = action?.['action-name'][0]?.text;
+  const actionName = action?.['action-contract-call'];
+  const chain = chainFromChainId[contractData?.chain_id];
+  const fiatCurrency = useFiatCurrency();
+  const chainRates = useChainRates(chain);
+  const currencySymbol = getCurrencySymbol(fiatCurrency);
+  const feeInfo = useRootSelector((root) => root.transactionEstimate.feeInfo);
+  const contractFunction = JSON.parse(contractData?.abi)?.find((fnRes) => fnRes.name === actionName);
+  const blueprint = action?.blueprint;
+  const sequence = blueprint ? JSON.parse(blueprint).sequence : null;
+  const blankArr = new Array(contractFunction?.inputs.length).fill(null);
+  const isPayable = contractType === 'payable' ? true : false;
+
+  const feeInFiat = getTxFeeInFiat(chain, feeInfo?.fee, feeInfo?.gasToken, chainRates, fiatCurrency);
+  const feeInFiatDisplayValue = `${currencySymbol}${feeInFiat.toFixed(5)}`;
+
+  const [value, setValue] = React.useState(blankArr);
+  const [contractRes, setContractRes] = React.useState([]);
+  const [integrationContract, setIntegrationContract]: any = React.useState();
+  const [isSendTransaction, setIsSendTransaction] = React.useState(false);
+  const [amount, setAmount] = React.useState(null);
+
+  React.useEffect(() => {
+    dispatch(fetchGasInfoAction(chain));
+    const integrationCon = etherspotService.getContract(chain, contractData?.abi, contractData?.contract_address);
+    setIntegrationContract(integrationCon);
+  }, [dispatch]);
+
+  const updateTxFee = async () => {
+    if (!value) return;
+    if (contractType === 'payable' ? (amount === null ? true : false) : false) return;
+    const findNull = value.find((res) => res === null || res === '');
+    if (findNull !== undefined) {
+      setContractRes(undefined);
+      return;
+    }
+    await contractInterface();
+  };
+
+  const updateTxFeeDebounced = React.useCallback(debounce(updateTxFee, 100), [value, amount]);
+
+  React.useEffect(() => {
+    updateTxFeeDebounced();
+    return updateTxFeeDebounced.cancel;
+  }, [updateTxFeeDebounced]);
+
+  const contractInterface = async () => {
+    const fnName = actionName ? `encode${actionName[0].toUpperCase()}${actionName.substring(1)}` : ``;
+    const updatedArr = value.map(async (specificVal, i) =>
+      specificVal?.c ? getDecimalValue(specificVal, i) : specificVal,
```

Comment:
`getDecimalValue` returns 0 when not found but i believe the ERC 20 standard is 18... @zincoshine could you confirm this please?

Reply:
This is return 0 when for store value and deadline. deadline and store value is working in only 0 decimal.